runtime/src/dispatch.rs

```diff
@@ -0,0 +1,180 @@
+use castaway::cast;
+use std::marker::PhantomData;
+
+use fvm_ipld_encoding::ipld_block::IpldBlock;
+use serde::{Deserialize, Serialize};
+
+use crate::ActorError;
+
+/// Implement actor method dispatch:
+///
+/// ```ignore
+/// type Actor;
+/// #[derive(FromPrimitive)]
+/// #[repr(u64)]
+/// enum Method {
+///     Constructor = 1,
+/// }
+/// impl ActorCode for Actor {
+///     type Methods = Method;
+///     actor_dispatch! {
+///         Constructor => constructor,
+///     }
+/// }
+/// ```
+#[macro_export]
+macro_rules! actor_dispatch {
+    ($($method:ident => $func:ident,)*) => {
+        fn invoke_method<RT>(
```

Comment:
maybe can change to `impl Runtime`

Reply:
We can't, I tried but is used as a parameter for an `FnOnce` and arguments of these need to be types.